styleguide-bash.md

```diff
@@ -0,0 +1,79 @@
+# installimage - Style Guide
+
+---
+
+## Contents
++ [Indentation Guidelines](#indentation-guidelines)
++ [Multiline Output to File](#multiline-output-to-file)
++ [Escaping](#escaping)
++ [Preferred Usage of Bash Builtins](#preferred-usage-of-bash-builtins)
++ [Inspiration](#inspiration)
+
+---
+
+## Indentation Guidelines
+we use two whitespaces for Indentation, and no hard tabs. This results in the following vim settings:
+```
+set tabstop=2
+set shiftwidth=2
+set expandtab
+set softtabstop=2
+```
+
+## Multiline Output to File
+Group the output of multiple commands with braces and redirect this once into a file. Here is a bad example:
+```bash
+echo "### $COMPANY - installimage" > "$CONFIGFILE"
+echo "# Loopback device:" >> "$CONFIGFILE"
+echo "auto lo" >> "$CONFIGFILE"
+echo "iface lo inet loopback" >> "$CONFIGFILE"
+echo "" >> "$CONFIGFILE"
+```
+
+This good example is:
+```bash
+{
+	echo "### $COMPANY - installimage"
+	echo "# Loopback device:"
+	echo "auto lo"
+	echo "iface lo inet loopback"
+	echo ""
+} > "$CONFIGFILE"
+```
+
+## Escaping
+We don't want dirty escaping for variables in `echo`, we should prefer printf in these cases, here is a bad example:
+```bash
+echo -e "SUBSYSTEM==\"net\", ACTION==\"add\", DRIVERS==\"?*\", ATTR{address}==\"$2\", ATTR{dev_id}==\"0x0\", ATTR{type}==\"1\", KERNEL==\"eth*\", NAME=\"$1\"" >> $UDEVFILE
```

Comment:
It is unfair to compare a double-quoted string containing double-quotes with a single-quoted string containing double-quotes.

Here is a better comparison:
```
echo 'SUBSYSTEM=="net", ACTION=="add", DRIVERS=="?*", ATTR{address}=="'"$2"'", ATTR{dev_id}=="0x0", ATTR{type}=="1", KERNEL=="eth*", NAME="'"$1"'"' >> $UDEVFILE
printf 'SUBSYSTEM=="net", ACTION=="add", DRIVERS=="?*", ATTR{address}=="%s", ATTR{dev_id}=="0x0", ATTR{type}=="1", KERNEL=="eth*", NAME="%s"\n' "$2" "$1" >> $UDEVFILE
```

In this case I would prefer the echo.